plugins/modules/kernel_blacklist.py

```diff
@@ -69,15 +69,23 @@ class Blacklist(StateModuleHelper):
     def __init_module__(self):
         self.pattern = re.compile(r'^blacklist\s+{0}$'.format(re.escape(self.vars.name)))
         self.vars.filename = self.vars.blacklist_file
-        self.vars.set('file_exists', os.path.exists(self.vars.filename), output=False, change=True)
-        if not self.vars.file_exists:
-            with open(self.vars.filename, 'a'):
-                pass
-            self.vars.file_exists = True
-            self.vars.set('lines', [], change=True, diff=True)
-        else:
-            with open(self.vars.filename) as fd:
-                self.vars.set('lines', [x.rstrip() for x in fd.readlines()], change=True, diff=True)
+        try:
+            dirpath = os.path.dirname(self.vars.filename)
+            if not os.path.isdir(dirpath):
+                self.module.fail_json(msg="The directory '{}' does not exist.".format(dirpath))
+            self.vars.set('file_exists', os.path.exists(self.vars.filename), output=False, change=True)
+            if not self.vars.file_exists:
+                with open(self.vars.filename, 'a'):
+                    pass
+                self.vars.file_exists = True
+                self.vars.set('lines', [], change=True, diff=True)
+            else:
+                with open(self.vars.filename) as fd:
+                    self.vars.set('lines', [x.rstrip() for x in fd.readlines()], change=True, diff=True)
+        except (OSError, IOError) as e:
+            self.module.fail_json(msg="Error accessing or creating blacklist file '{}': {}".format(self.vars.filename, e))
```

Comment:
It's better to let Python do the quoting:
```suggestion
            self.module.fail_json(msg="Error accessing or creating blacklist file {!r}: {}".format(self.vars.filename, e))
```
Same below.